File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function solution(n, k) {
  let answer = 0;
  const binary = n.toString(k);

  const discriminateDemical = (binary) => {
    if (binary == 2) {
      return true;
    }
    if (binary == 1) {
      return false;
    }
    for (let i = 2; i <= Math.floor(Math.sqrt(binary)); i++) {
      if (binary % i === 0) {
        return false;
      }
    }
    return true;
  };

  const splitZero = binary.split("0");

  for (let i = 0; i < splitZero.length; i++) {
    if (!splitZero[i]) {
      continue;
    }
    const isDemical = discriminateDemical(splitZero[i]);
    if (isDemical) {
      answer++;
    }
  }

  return answer;
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        String binaryNumber = Integer.toString(n, k);
        String[] splitNumber = binaryNumber.split("0");

        for (String number : splitNumber) {
            if (!number.isBlank() && isPrime(Long.parseLong(number))) answer++;
        }

        return answer;
    }

    private boolean isPrime(Long number) {
        if (number <= 1) {
            return false;
        }

        for (int i = 2; i < (int) Math.sqrt(number) + 1; i++) {
            if (number % i == 0) {
                return false;
            }
        }

        return true;
    }
}